src/query/interpreter.cpp

```diff
@@ -4016,6 +4016,136 @@ PreparedQuery PrepareCreateSnapshotQuery(ParsedQuery parsed_query, bool in_expli
+  auto evaluator = PrimitiveLiteralExpressionEvaluator{evaluation_context};
+
+  return PreparedQuery{
+      {},
+      std::move(parsed_query.required_privileges),
+      [storage, replication_role, path = recover_query->snapshot_->Accept(evaluator).ValueString(),
+       force = recover_query->force_](AnyStream * /*stream*/,
+                                      std::optional<int> /*n*/) -> std::optional<QueryHandlerResult> {
+        auto *mem_storage = static_cast<storage::InMemoryStorage *>(storage);
+        if (auto maybe_error = mem_storage->RecoverSnapshot(path, force, replication_role); maybe_error.HasError()) {
+          switch (maybe_error.GetError()) {
+            case storage::InMemoryStorage::RecoverSnapshotError::DisabledForReplica:
+              throw utils::BasicException(
+                  "Failed to recover a snapshot. Replica instances are not allowed to create them.");
+            case storage::InMemoryStorage::RecoverSnapshotError::DisabledForMainWithReplicas:
+              throw utils::BasicException(
+                  "Failed to recover a snapshot. Cannot recover if instance has registered replicas.");
+            case storage::InMemoryStorage::RecoverSnapshotError::NonEmptyStorage:
+              throw utils::BasicException("Failed to recover a snapshot. Storage is not clean. Try using FORCE.");
+            case storage::InMemoryStorage::RecoverSnapshotError::MissingFile:
+              throw utils::BasicException("Failed to find the defined snapshot file.");
+            case storage::InMemoryStorage::RecoverSnapshotError::CopyFailure:
+              throw utils::BasicException("Failed to copy snapshot over to local snapshots directory.");
+            case storage::InMemoryStorage::RecoverSnapshotError::BackupFailure:
+              throw utils::BasicException(
+                  "Failed to clear local wal and snapshots directories. Please clean them manually.");
+          }
+        }
+        return QueryHandlerResult::COMMIT;
+      },
+      RWType::NONE};
+}
+
+PreparedQuery PrepareShowSnapshotsQuery(ParsedQuery parsed_query, bool in_explicit_transaction, CurrentDB &current_db) {
+  if (in_explicit_transaction) {
+    throw ShowSchemaInfoInMulticommandTxException();
+  }
+
+  MG_ASSERT(current_db.db_acc_, "Show Snapshots query expects a current DB");
+  storage::Storage *storage = current_db.db_acc_->get()->storage();
+
+  if (storage->GetStorageMode() == storage::StorageMode::ON_DISK_TRANSACTIONAL) {
+    throw ShowSnapshotsDisabledOnDiskStorage();
+  }
+
+  Callback callback;
+  callback.header = {"path", "timestamp", "creation_time", "size"};
+  callback.fn = [storage]() mutable -> std::vector<std::vector<TypedValue>> {
+    std::vector<std::vector<TypedValue>> infos;
+    const auto res = static_cast<storage::InMemoryStorage *>(storage)->ShowSnapshots();
+    infos.reserve(res.size());
+
+    auto readable_size = [](size_t size) {
```

Comment:
I think you already have a function for that utils::GetReadableSize

Reply:
didn't know that existed :D
cleaner also